web/src/components/tasks/CannabisPriorityStatus.tsx

```diff
@@ -0,0 +1,342 @@
+import { Content } from "@/components/Content";
+import { Button } from "@/components/njwds-extended/Button";
+import { ToastAlert } from "@/components/njwds-extended/ToastAlert";
+import { TaskHeader } from "@/components/TaskHeader";
+import { useTaskFromRoadmap } from "@/lib/data-hooks/useTaskFromRoadmap";
+import { useUserData } from "@/lib/data-hooks/useUserData";
+import { CannabisPriorityStatusDisplayContent, Task } from "@/lib/types/types";
+import { scrollToTop } from "@/lib/utils/helpers";
+import { getUnlockedBy } from "@/pages/tasks/[taskUrlSlug]";
+import Config from "@businessnjgovnavigator/content/fieldConfig/config.json";
+import { Checkbox, FormControlLabel } from "@mui/material";
+import React, { useEffect, useMemo, useState } from "react";
+
+interface Props {
+  task: Task;
+  displayContent: CannabisPriorityStatusDisplayContent;
+}
+
+/**
+ *
+ *
+ * Need to discuss placement of this object and creation of CMS collection
+ *
+ */
+export const priorityTypesObj = {
+  minorityOrWomen: ["general-minority-owned", "general-women-owned"],
+  veteran: ["general-veteran-owned"],
+  impactZone: [
+    "cannabis-business-in-impact-zone",
+    "cannabis-owner-in-impact-zone",
+    "cannabis-employee-in-impact-zone",
+  ],
+  socialEquity: [
+    "cannabis-economically-disadvantaged-social-equity",
+    "cannabis-criminal-offense-social-equity",
+  ],
+  noneOfTheAbove: "cannabis-priority-status-none",
+};
+
+export const CannabisPriorityStatus = (props: Props) => {
+  const { userData, update } = useUserData();
+  const taskFromRoadmap = useTaskFromRoadmap(props.task.id);
+  const unlockedByTask = getUnlockedBy(taskFromRoadmap, userData);
+  const [successToastIsOpen, setSuccessToastIsOpen] = useState(false);
+
+  const [displayFirstTab, setDisplayFirstTab] = useState(true);
+  const [displayNextTabButton, setDisplayNextTabButton] = useState(false);
+
+  const [displayMWPriorityType, setDisplayMWPriorityType] = useState(false);
+  const [displayVeteranPriorityType, setDisplayVeteranPriorityType] = useState(false);
+  const [displayImpactZonePriorityType, setDisplayImpactZonePriorityType] = useState(false);
+  const [displaySocialEquityPriorityType, setDisplaySocialEquityPriorityType] = useState(false);
+  const [displayNoPriorityType, setDisplayNoPriorityType] = useState(false);
+
+  const priorityTypes = useMemo(() => {
+    return [
+      ...priorityTypesObj.minorityOrWomen,
+      ...priorityTypesObj.veteran,
+      ...priorityTypesObj.impactZone,
+      ...priorityTypesObj.socialEquity,
+    ];
+  }, []);
+
+  useEffect(() => {
+    if (userData && userData.taskItemChecklist && displayFirstTab) {
```

Comment:
why the check for `userData.taskItemChecklist` - that should never be undefined, right? only ever `{}`